pkgs/by-name/li/libx11/package.nix

```
@@ -0,0 +1,100 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  buildPackages,
+  pkg-config,
+  xorgproto,
+  libxcb,
+  xtrans,
+  writeScript,
+  testers,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libx11";
+  version = "1.8.12";
+
+  outputs = [
+    "out"
+    "dev"
+    "man"
+  ];
+
+  src = fetchurl {
+    url = "mirror://xorg/individual/lib/libX11-${finalAttrs.version}.tar.xz";
+    hash = "sha256-+gJvm7AST01sgI+a70BXqtZeezXY/0OVHO8Kvga7mpo=";
+  };
+
+  strictDeps = true;
+
+  depsBuildBuild = [
+    buildPackages.stdenv.cc
+  ] ++ lib.optional stdenv.hostPlatform.isStatic (buildPackages.stdenv.cc.libc.static or null);
```

This seems odd to me, because we're checking the host platform but then adding inputs for the build platform. I know it was previously xorg.buildPackages.stdenv, so maybe buildPackages.stdenv is something else? Have you tried a static build? I'd expect it to be 

```suggestion
  ] ++ lib.optional stdenv.hostPlatform.isStatic (stdenv.cc.libc.static or null);
```

---

Wouldn't `depsBuildBuild` have to come from `pkgsBuildBuild` and not `buildPackages` which according to the nixpkgs-manual is an alias to `pkgsBuildHost`?

---

this just gave me a major brainfuck so i asked on discourse https://discourse.nixos.org/t/why-is-buildpackages-aka-pkgsbuildhost-used-in-depsbuildbuild-instead-of-pkgsbuildbuild/65334

---

found it `nix build .#pkgsStatic.libx11`

---

since `xorg.buildPackages` doesn't exist, `xorg.buildPackages.libc.static or null` evaluates to `null`
so the the line didn't do anything

adding `libc.static` actually breaks the static build in a way i can't explain